compiler/src/compile.rs

```diff
@@ -1303,11 +1303,9 @@ impl<O: OutputStream> Compiler<O> {
                             self.emit(Instruction::Pop);
```

Comment:
I would expect the same change to be required for this line. Still I do not fully grasp the problem.